config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/FolderEntityBuilder.java

```diff
@@ -44,14 +47,24 @@ private Entity buildFolderEntity(Folder folder, String id, String parentFolderId
         if (parentFolderId != null) {
             folderElement.setAttribute(ATTRIBUTE_FOLDER_ID, parentFolderId);
         }
-        String filteredName = CharacterBlacklistUtil.filterAndReplace(folder.getName());
-        folderElement.appendChild(createElementWithTextContent(document, NAME, filteredName));
+        String folderName = folder.getName();
+        try {
+            folderName = CharacterBlacklistUtil.decodeName(folderName);
+        } catch (UnsupportedEncodingException e) {
+            LOGGER.log(Level.WARNING, "unable to decode folder name " + folderName);
```

Comment:
@uppoju , if we are not able to decode the folder name, shouldn't we throw the exception? if bundle is generated with encoded name, will it not be a problem?